src/Core/Suite.php

```diff
@@ -111,4 +127,23 @@ protected function runTest(TestInterface $test, TestResult $result)
         $test->setEventEmitter($this->eventEmitter);
         $test->run($result);
     }
+
+    private function focusedTests()
```

Comment:
Couple thoughts on this:

I don't think I ever published any real coding standards, so I am open to being shot down on this :) - but for this project (and generally in the PHP realm for me) - I tended more towards the `get` language for things returning something. It is a little more verbose, but I think it fits the language and the rest of the project a little better. Something like `getFocusedTests`. In JavaScript and other functional languages I use, I like to ditch some of that verbosity. Thoughts?

Second, I'm not sure highlighting `focused` here is the best solution. It is currently our only need so again I'm probably wrong :). It kind of seems like we are applying a transformation of state to the test tree before we run things, and this seems like something that might fit well into an `array_filter` type scenario:

```php
// or maybe getTestsToRun?
protected function getFilteredTests()
{
    $tests = array_filter($this->tests, function (TestInterface $test) {
        return $test->getFocused();
    });

    return empty($tests) ? $this->tests : $tests;
}
```

Basically, if the state of the things to run has changed we know we can return that changed set. Otherwise return the original. The language might be a bit more flexible if we add similar functionality later on, and it also eliminates the need for the additional `$hasFocusedTests` variable. Thoughts?